### PR TITLE
Audit Log Usage - Content API - Async logging #7627

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

**Comment by vbradnitski** [comment]
should be merged after #7616

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ContentAuditLogSupport.java [hunk 1]
```diff
@@ -42,346 +55,531 @@
...
 import com.enonic.xp.security.auth.AuthenticationInfo;
 import com.enonic.xp.site.CreateSiteParams;
+import com.enonic.xp.site.Site;
 
 class ContentAuditLogSupport
 {
-
     private static final String SOURCE_CORE_CONTENT = "com.enonic.xp.core-content";
 
+    private static final AtomicBoolean initialized = new AtomicBoolean( false );
```
Comment (jsi) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ContentAuditLogSupport.java]: Codacy Issue found: [Name 'initialized' must match pattern '^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$'.](https://app.codacy.com/app/enonic/xp/pullRequest?prid=4499884)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- modules/lib/lib-content/src/main/java/com/enonic/xp/lib/content/mapper/PublishContentResultMapper.java  3
- modules/core/core-api/src/main/java/com/enonic/xp/audit/LogAuditLogParams.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/content/PublishContentException.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/content/PublishContentParams.java  4
         

Complexity decreasing per file
==============================
+ modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ContentAuditLogSupport.java  -2
         

Clones removed
==============
+ modules/core/core-api/src/main/java/com/enonic/xp/audit/LogAuditLogParams.java  -1
+ modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/PublishContentCommand.java  -2
+ modules/core/core-api/src/main/java/com/enonic/xp/audit/AuditLog.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/xp/xp/pullRequest?prid=4499884&bid=15169678)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-api/src/main/java/com/enonic/xp/audit/LogAuditLogParams.java  1
- modules/lib/lib-content/src/main/java/com/enonic/xp/lib/content/mapper/PublishContentResultMapper.java  3
- modules/core/core-api/src/main/java/com/enonic/xp/content/PublishContentParams.java  4
- modules/core/core-api/src/main/java/com/enonic/xp/content/PublishContentException.java  1
         

Complexity decreasing per file
==============================
+ modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ContentAuditLogSupport.java  -2
         

Clones removed
==============
+ modules/core/core-api/src/main/java/com/enonic/xp/audit/LogAuditLogParams.java  -1
+ modules/core/core-api/src/main/java/com/enonic/xp/audit/AuditLog.java  -1
+ modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/PublishContentCommand.java  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/xp/xp/pullRequest?prid=4499884&bid=15169678)

**Comment by GlennRicaud** [comment]
Could you revert the renaming of params and listener. This is breaking

**Comment by GlennRicaud** [comment]
Nice use of new ThreadPoolExecutor.CallerRunsPolicy()
But if will not be called if you do not set a limit to the queue.

Also this get/set is technically a race condition (Even if it will not happen).

**Pull request merged**